src/journal/journalctl.c

```diff
@@ -1930,6 +1930,10 @@ static int simple_varlink_call(const char *option, const char *method) {
                 "/run/systemd/journal/io.systemd.journal";
 
         r = varlink_connect_address(&link, fn);
+        if (r == -ECONNREFUSED && arg_namespace && streq(method, "io.systemd.Journal.Synchronize"))
```

Comment:
Hmm, I'd prefer that we do such checks for specific methods in their own functions. I.e. let's rename this function to `varlink_connect_journal`, and leave `varlink_call_and_log` for callers. We don't have to do almost pointless string comparisons in that way,

Reply:
Makes sense. I actually thought about doing that too, since it definitely feels icky this way.